docs/plugin-writer/tasks/add-remove.rst

```diff
@@ -1,8 +1,27 @@
 Adding and Removing Content
 ===========================
 
-TODO: RepositoryVersion
-https://pulp.plan.io/issues/3220
+Repositories have versions. A new respository version is created when its set of content units
+changes.
+
+To facilitate the creation of repository versions a :class:`~pulpcore.plugin.models.RepositoryVersion`
+context manager is provided. Plugin Writers are strongly encouraged to use RepositoryVersion as
+a context manager to provide transactional safety, working directory set up, and cleaning
+up the database on failures::
+
+     with RepositoryVersion.create(repository) as new_version:
```

Comment:
Maybe add highlighting (ie `.. code-block:: python`)?

Reply:
I think this is redundant. we should just link to this code block in the template.